qiskit_nature/utils/linalg.py

```diff
@@ -179,3 +179,136 @@ def _swap_columns(matrix: np.ndarray, i: int, j: int) -> None:
     column_i = matrix[:, i].copy()
     column_j = matrix[:, j].copy()
     matrix[:, i], matrix[:, j] = column_j, column_i
+
+
+def modified_cholesky(
+    mat: np.ndarray, *, error_threshold: float = 1e-8, max_vecs: int | None = None
+) -> np.ndarray:
+    r"""Modified Cholesky decomposition.
+
+    The modified Cholesky decomposition of a square matrix :math:`M` has the form
+
+    .. math::
+        M = \sum_{i=1}^N v_i v_i^\dagger
+
+    where each :math:`v_i` is a vector. :math:`M` must be positive definite.
+    No checking is performed to verify whether :math:`M` is positive definite.
+    The number of terms :math:`N` in the decomposition depends on the allowed
+    error threshold. A larger error threshold may yield a smaller number of terms.
+    Furthermore, the ``max_vecs`` parameter specifies an optional upper bound
+    on :math:`N`. The ``max_vecs`` parameter is always respected, so if it is
+    too small, then the error of the decomposition may exceed the specified
+    error threshold.
+
+    .. note::
+        No checking is performed to verify whether the input matrix is positive definite.
+        If the input matrix is not positive definite, then the decomposition returned will be invalid.
```

Comment:
Just saw this now. Silently "working" seems like a bad idea, especially given that checking this matrix is fairly simple. I suggest we add a check which can be disabled via a keyword argument if you really do not want this to be checked.

Reply:
Sorry, I had misread the condition there as something that could be easily checked :sweat_smile: I suggest changing the admonition from `note` to `warning` to emphasize this a bit more.

Reply:
Done.